bin/reth/src/cli.rs

```diff
@@ -80,6 +92,35 @@ struct Cli {
     #[clap(subcommand)]
     command: Commands,
 
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    ///
+    /// Built-in chains:
+    /// - mainnet
+    /// - goerli
+    /// - sepolia
+    #[arg(
+        long,
+        value_name = "CHAIN_OR_PATH",
+        global = true,
+        verbatim_doc_comment,
+        default_value = "mainnet",
+        value_parser = genesis_value_parser,
+        global = true,
+    )]
+    chain: Arc<ChainSpec>,
+
+    /// The path to the data dir for all reth files and subdirectories.
+    ///
+    /// Defaults to the OS-specific data directory:
+    ///
+    /// - Linux: `$XDG_DATA_HOME/reth/` or `$HOME/.local/share/reth/`
+    /// - Windows: `{FOLDERID_RoamingAppData}/reth/`
+    /// - macOS: `$HOME/Library/Application Support/reth/`
+    #[arg(long, value_name = "DATA_DIR", global = true, verbatim_doc_comment, default_value_t)]
+    datadir: MaybePlatformPath<DataDirPath>,
```

Comment:
I would remove this, and just use the `Logs` directory, creating chain specific subdirectories

Reply:
I actually was thinking about this when solving the issue but at the end I chose to follow strictly what was written in the issue.

I believe it makes more sense to directly use datadir for Logs and get rid of logs directory. So that every time you set up a Reth node you know that you have all the relevant data in a specific folder, logs included.

The first implementation of this I have in mind is to leave the datadir (and of course the log field) in the Cli struct but deleting the if else branch I created and always move logs to the datadir folder.

Something like this:
```rust
struct Cli {
    /// The command to run
    #[clap(subcommand)]
    command: Commands,

    /// The path to the data dir for all reth files and subdirectories.
    ///
    /// Defaults to the OS-specific data directory:
    ///
    /// - Linux: `$XDG_DATA_HOME/reth/` or `$HOME/.local/share/reth/`
    /// - Windows: `{FOLDERID_RoamingAppData}/reth/`
    /// - macOS: `$HOME/Library/Application Support/reth/`
    #[arg(long, value_name = "DATA_DIR", global = true, verbatim_doc_comment, default_value_t)]
    datadir: MaybePlatformPath<DataDirPath>,

    #[clap(flatten)]
    logs: Logs,

    #[clap(flatten)]
    verbosity: Verbosity,
}

/// Parse CLI options, set up logging and run the chosen command.
pub fn run() -> eyre::Result<()> {
    let mut opt = Cli::parse();
    
    // move logs to the already-chain-specific --datadir
    opt.logs.log_directory = opt.datadir.unwrap_or_default().map_to::<LogsDir>();

    ...
}

```

This way we don't have to change the `layer` method of `Logs` and also we make clear that `logs_directory` is a different thing then `datadir` also if they are the same value. 
Also we can delete the `chain` field in the Cli struct since it is not necessary anymore.

What do you think of this approach? Do you think it's a valid solution or you'd like to delete the `log_directory` folder and change the `layer` method of `Logs` completely?